### PR TITLE
Add Coveralls GitHub Action

### PR DESCRIPTION
**Pull request by michplunkett**
## Fixes issue
https://github.com/lucyparsons/OpenOversight/issues/914
https://github.com/lucyparsons/OpenOversight/issues/937

## Description of Changes
This PR creates a Coveralls GitHub action that sends our code coverage to its respective portal and fixes the broken link that was in the `README`.

## Notes for Deployment
- @abandoned-prototype will re-integrate Coveralls with the OO repository.

## Tests and linting
 - [x] This branch is up-to-date with the `develop` branch.
 - [x] `pytest` passes on my local development environment.
 - [x] `pre-commit` passes on my local development environment.

### CODE REVIEW

**Comment by michplunkett** [comment]
The good news is that our builds are now showing up on Coveralls again and the bad news is that we aren't sending anything meaningful at this point. :( 
Screenshot 2023-06-26 at 5 45 28 PM

Altogether though, this means we're like 99% of the way there!

**Comment by michplunkett** [comment]
```bash
Run coverallsapp/github-action@master
  with:
    github-token: ***
    path-to-lcov: OpenOversight/tests/coverage.xml
    coveralls-endpoint: https://coveralls.io/
    debug: false
  env:
    FLASK_APP: OpenOversight.app
Using lcov file: OpenOversight/tests/coverage.xml
Error: ENOENT: no such file or directory, open 'OpenOversight/tests/coverage.xml'
```

Here's the error I'm ending the day on. I'm gettin' REAL CLOSE.

**Comment by abandoned-prototype** [comment]
Oops, I wanted to test the stuff in my own branch, but I added my commit here by mistake. Anyways, it looks like with just one small change it works. Great work, thank you!

File: Makefile [hunk 1]
```diff
@@ -48,9 +48,9 @@ populate: create_db  ## Build and run containers
 .PHONY: test
 test: start  ## Run tests
 	if [ -z "$(name)" ]; then \
-		FLASK_ENV=testing docker-compose run --rm web pytest --doctest-modules -n auto --dist=loadfile -v OpenOversight/tests/; \
+		FLASK_ENV=testing docker-compose run --rm web pytest --cov=OpenOversight --cov-report xml:OpenOversight/tests/coverage.xml --doctest-modules -n auto --dist=loadfile -v OpenOversight/tests/; \
```
Comment (abandoned-prototype) [inline Makefile:51]: These are only used for local testing, but it seems fine to generate coverage reports here as well

**Review by abandoned-prototype — APPROVED** [review]
Feel free to merge if this looks good to you

File: Makefile [hunk 2]
```diff
@@ -48,9 +48,9 @@ populate: create_db  ## Build and run containers
 .PHONY: test
 test: start  ## Run tests
 	if [ -z "$(name)" ]; then \
-		FLASK_ENV=testing docker-compose run --rm web pytest --doctest-modules -n auto --dist=loadfile -v OpenOversight/tests/; \
+		FLASK_ENV=testing docker-compose run --rm web pytest --cov --doctest-modules -n auto --dist=loadfile -v OpenOversight/tests/; \
 	else \
-	    FLASK_ENV=testing docker-compose run --rm web pytest --doctest-modules -v OpenOversight/tests/ -k $(name); \
+	    FLASK_ENV=testing docker-compose run --rm web pytest --cov --doctest-modules -v OpenOversight/tests/ -k $(name); \
```
Comment (michplunkett) [inline Makefile:53]: I don't think we need the coverage files for local testing, but the coverage statistics are helpful to have when running `make test`.

File: .github/workflows/test_coverage.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+name: Test Coverage
+on:
+  push:
+    branches:
+      - develop
+      - main
```
Comment (michplunkett) [inline .github/workflows/test_coverage.yml:6]: Both other yml files callout develop and main, so I figured it'd be good for uniformity to do the same within this file as well.

File: .github/workflows/test_prs.yml [hunk 4]
```diff
@@ -4,8 +4,8 @@ name: CI
 on:
   pull_request:
     branches:
-    - develop
-    - main
+      - develop
+      - main
```
Comment (michplunkett) [inline .github/workflows/test_prs.yml:8]: Consistent spacing between `yml` files.

File: .github/workflows/test_coverage.yml (hunk 3 above)
Comment (abandoned-prototype) [reply .github/workflows/test_coverage.yml]: I reverted this change because we only track coverage on the develop branch.

**Pull request merged**